amaranth_soc/csr/field.py

```diff
@@ -0,0 +1,188 @@
+from amaranth import *
+
+from .reg import Field
+
+
+__all__ = ["R", "W", "RW", "RW1C", "RW1S"]
+
+
+class R(Field):
+    __doc__ = Field._doc_template.format(
+    description="""
+    A read-only field.
+    """.strip(),
+    parameters="",
+    attributes="""
+    r_data : Signal(shape)
+        Read data. Drives the :attr:`~FieldPort.r_data` signal of ``port``.
+    """.strip())
+
+    def __init__(self, shape):
+        super().__init__(shape, access="r")
+        self.r_data = Signal(shape)
+
+    def elaborate(self, platform):
+        m = Module()
+        m.d.comb += self.port.r_data.eq(self.r_data)
+        return m
+
+
+class W(Field):
+    __doc__ = Field._doc_template.format(
+    description="""
+    A write-only field.
+    """.strip(),
+    parameters="",
+    attributes="""
+    w_data : Signal(shape)
+        Write data. Driven by the :attr:`~FieldPort.w_data` signal of ``port``.
```

Comment:
Would be good to forward `port.w_stb` too and clarify that `w_data` is only valid on cycles where `w_stb` is asserted, if the intention is indeed to have a field with no storage.

Reply:
IIRC we've discussed this with @jfng and the intent was to have only `w_stb` on the register, not on the field, to avoid having two similar ways to do the same thing. An earlier design would have been able to assert writes on only some fields but we got rid of that and decided to make it explicit.

Reply:
Then I think the attribute description should be clarified.